src/main/java/io/supertokens/webserver/api/core/DeleteUserAPI.java

```diff
@@ -45,15 +48,22 @@ public String getPath() {
 
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        // this API is app specific
         JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
         String userId = InputParser.parseStringOrThrowError(input, "userId", false);
         try {
-            AuthRecipe.deleteUser(this.getTenantIdentifier(req), super.main, userId);
-            JsonObject result = new JsonObject();
-            result.addProperty("status", "OK");
-            super.sendJsonResponse(200, result, resp);
+            AppIdentifierStorageAndUserIdMapping appIdentifierStorageAndUserIdMapping =
+                    this.getAppIdentifierStorageAndUserIdMapping(req, userId, UserIdType.SUPERTOKENS);
```

Comment:
why are you doing user id mapping with supertokens user id type here? The AuthRecipe.deleteUser does user id mapping check with ANY in it anyway. So what you have done here is wrong

Reply:
Done